docs/src/main/asciidoc/cdi-integration.adoc

```diff
@@ -0,0 +1,606 @@
+_Bean discovery_ is then the process where the container analyzes all application classes, identifies beans and wires them all together based on the provided metadata.
+Subsequently, the extensions can register _synthetic components_.
+Attributes of these components are fully contolled by the extensions, i.e. are not derived from an existing class.
+Finally, the _deployment is validated_.
+For example, the container validates every injection point in the application and fails the build if there is no bean that satisfies the given required type and qualifiers.
+
+TIP: You can see more information about the bootstrap by enabling additional logging. Simply run the Maven build with `-X,--debug` and grep lines that contain `io.quarkus.arc`. In the <<cdi-reference.adoc#dev-mode,development mode>>, you can use `quarkus.log.category."io.quarkus.arc.processor".level=DEBUG` and two special endpoints are also registered automatically to provide some basic debug info in the JSON format.
+
+Quarkus build steps can produce and consume various build items and hook into each phase.
+In the following sections we will describe all the relevant build items and common scenarios.
+
+[IMPORTANT]
+.Metadata Sources
+====
+Classes and annotations are the primary source of bean-level metadata.
+However, extensions can add, remove or transform the metadata at certain stages of the bootstrap.
+Moreover, extenions can also register <<synthetic_beans,synthetic components>>. 
+This is an important aspect to realize when integrating CDI components in Quarkus.
+====
+
+== Use Case - My Class Is Not Recognized as a Bean
+
+An `UnsatisfiedResolutionException` indicates a problem during <<cdi.adoc#typesafe_resolution,typesafe resolution>>.
+Sometimes an injection point cannot be satisifed even if there is a class on the class path that appears to be eligible for injection.
+There are several reasons why a class is not recognized and also several ways to fix it.
+In the first step we should identify the _reason_.
+
+[[additional_bean_build_item]]
+=== _Reason 1_: Class Is Not discovered 
+
+Quarkus has a <<cdi-reference.adoc#bean_discovery,simplified discovery>>.
+It might happen that the class is not part of the application index.
+For example, classes from the _runtime module_ of a Quarkus extension are not indexed automatically.
+
+_Solution_: Use the `AdditionalBeanBuildItem`. 
+This build item can be used to specify one or more additional classes to be analyzed during the discovery. 
+Additional bean classes are transparently added to the application index processed by the container.
+
+.`AdditionalBeanBuildItem` Example
+[source,java]
+----
+@BuildStep
+AdditionalBeanBuildItem additionalBeans() {
+     return new AdditionalBeanBuildItem(SmallRyeHealthReporter.class, HealthServlet.class)); <1>
+}
+----
+<1> Convenient `AdditionalBeanBuildItem.Builder` can be used for more complex use cases.
```

Comment:
I think you can drop `convenient` here - it doesn't add anything to the sentence